### PR TITLE
book feature

### PR DESCRIPTION
**Pull request by Alastair-Revell**

### CODE REVIEW

File: lib/space.rb [hunk 1]
```diff
@@ -22,6 +24,11 @@ def self.add(name:, description:, price:, date:)
     Space.create(name: name, description: description, price: price, date: date)
   end
 
+  def book(name)
+    entry = Space.first(:name => name)
```
Comment (sarah-byrne) [inline lib/space.rb:28]: is name unique?

File: spec/space_spec.rb [hunk 2]
```diff
@@ -8,4 +8,11 @@
     expect(Space[1][:name]).to eq("Fourth space")
   end
 
+  it "can update booked status" do
+    setup
+    entry = Space.first(:name => 'Third space')
+    entry.book("Third space")
```
Comment (sarah-byrne) [inline spec/space_spec.rb:14]: Would it be better to use id to update?

File: lib/space.rb [hunk 3]
```diff
@@ -22,6 +24,11 @@ def self.add(name:, description:, price:, date:)
...
+    entry.update(:available => false)
+  end
+
 end
 
 DataMapper.finalize
```
Comment (sarah-byrne) [inline lib/space.rb:34]: what does finalize do?

File: lib/space.rb (hunk 1 above)
Comment (Alastair-Revell) [reply lib/space.rb:28]: It is possible that name could be duplicated in the table, if two listed properties have the same name. There are a couple of ways we could fix this.

File: lib/space.rb (hunk 3 above)
Comment (Alastair-Revell) [reply lib/space.rb:34]: It is called after all the models have been defined. It does a basic validity check of the DataMapper models

File: spec/space_spec.rb (hunk 2 above)
Comment (sarah-byrne) [reply spec/space_spec.rb:14]: task added to backlog to use id

File: lib/space.rb (hunk 1 above)
Comment (sarah-byrne) [reply lib/space.rb:28]: task added to backlog to use id

**Comment by sarah-byrne** [comment]
:shipit:

**Pull request merged**